### PR TITLE
feat: add kotlin module

### PR DESCRIPTION
**Pull request by SubSide**

### CODE REVIEW

File: kotlin/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    compileOnly(project(":core"))
+
+    // Kotlin dependencies
+    api(group = "org.jetbrains.kotlinx", name = "kotlinx-coroutines-core", version = versionKotlin)
+
+    // Tests
+    testImplementation(project(":core"))
+    testImplementation(project(":handler-simple"))
+    testImplementation(testFixtures(project(":core")))
+    testImplementation(group = "org.jetbrains.kotlinx", name="kotlinx-coroutines-test", version = versionKotlin)
```
Comment (PhilippHeuer) [inline kotlin/build.gradle.kts:15]: I think we only need a dependency on the api module for kotlin, there is no need for compileOnly unless i'm missing something.

```suggestion
    // project
    api(project(":api"))
    testImplementation(project(":core"))
    testImplementation(project(":handler-simple"))
    testImplementation(testFixtures(project(":core")))

    // kotlin
    api(group = "org.jetbrains.kotlinx", name = "kotlinx-coroutines-core", version = versionKotlin)
    testImplementation(group = "org.jetbrains.kotlinx", name="kotlinx-coroutines-test", version = versionKotlin)
```

**Review by PhilippHeuer — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution, this looks good. 🚀

I would just do one minor change in the `build.gradle.kts`

File: kotlin/build.gradle.kts (hunk 1 above)
Comment (SubSide) [reply kotlin/build.gradle.kts:15]: Good call!

Comment (SubSide) [reply kotlin/build.gradle.kts]: I also inlined versionKotlin. Since we're not sure if dependabot picks up variables

**Review by PhilippHeuer — APPROVED** [review]

**Pull request merged**